Sources/Fuzzilli/Mutators/OperationMutator.swift

```diff
@@ -119,22 +119,72 @@ public class OperationMutator: BaseInstructionMutator {
             newOp = ReassignWithBinop(chooseUniform(from: allBinaryOperators))
         case let op as DestructArray:
             var newIndices = Set(op.indices)
-            if newIndices.count > 0 {
-                newIndices.remove(newIndices.randomElement()!)
-                while newIndices.count != op.indices.count {
-                    newIndices.insert(Int.random(in: 0..<10))
+            var removedIndex = -1
```

Comment:
Ok cool, this logic makes sense to be, but I'm wondering if we can factor it out instead of essentially duplicating it four times.

How about something like this (with some slight simplifications as well):
```swift
        func replaceRandomElement<T>(in set: inout Set<T>, generatingRandomValuesWith generator: () -> T) {
            guard let removedElem = set.randomElement() else { return }
            set.remove(removedElem)

            for _ in 0...5 {
                let newElem = generator()
                // Ensure that we neither add an element that already exists nor add one that we just removed
                if !set.contains(newElem) && newElem != removedElem {
                    set.insert(newElem)
                    break
                }
            }
        }
```

then call it like
```swift
var newIndices = Set(op.indices)
replaceRandomElement(in: &newIndices, generatingRandomValuesWith: { return Int.random(in 0..<10) })
```

Do you think that'd work?

Reply:
Changes added, I've replaced the guard statement with and if-else block. If we can't remove an element, say if the set is empty, then instead of bailing we should attempt to add a new property.